// @aws-sdk/client-glue/dist-es/commands/ListWorkflowsCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListWorkflowsRequestFilterSensitiveLog, ListWorkflowsResponseFilterSensitiveLog, } from "../models/models_2";
import { deserializeAws_json1_1ListWorkflowsCommand, serializeAws_json1_1ListWorkflowsCommand, } from "../protocols/Aws_json1_1";
export class ListWorkflowsCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, ListWorkflowsCommand.getEndpointParameterInstructions()));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "GlueClient";
        const commandName = "ListWorkflowsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: ListWorkflowsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: ListWorkflowsResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_1ListWorkflowsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_1ListWorkflowsCommand(output, context);
    }
}
