// @aws-sdk/client-glue/dist-cjs/models/models_1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUserDefinedFunctionRequestFilterSensitiveLog = exports.UserDefinedFunctionInputFilterSensitiveLog = exports.ResourceUriFilterSensitiveLog = exports.CreateTriggerResponseFilterSensitiveLog = exports.CreateTriggerRequestFilterSensitiveLog = exports.CreateTableResponseFilterSensitiveLog = exports.CreateTableRequestFilterSensitiveLog = exports.TableInputFilterSensitiveLog = exports.TableIdentifierFilterSensitiveLog = exports.CreateSessionResponseFilterSensitiveLog = exports.SessionFilterSensitiveLog = exports.CreateSessionRequestFilterSensitiveLog = exports.SessionCommandFilterSensitiveLog = exports.CreateSecurityConfigurationResponseFilterSensitiveLog = exports.CreateSecurityConfigurationRequestFilterSensitiveLog = exports.EncryptionConfigurationFilterSensitiveLog = exports.S3EncryptionFilterSensitiveLog = exports.JobBookmarksEncryptionFilterSensitiveLog = exports.CloudWatchEncryptionFilterSensitiveLog = exports.CreateScriptResponseFilterSensitiveLog = exports.CreateScriptRequestFilterSensitiveLog = exports.FilterOperator = exports.FieldName = exports.PermissionTypeMismatchException = exports.PermissionType = exports.StatementState = exports.SchemaDiffType = exports.PartitionIndexStatus = exports.BackfillErrorCode = exports.TransformSortColumnType = exports.TransformStatusType = exports.SortDirectionType = exports.TaskRunSortColumnType = exports.TaskType = exports.CatalogEncryptionMode = exports.ResourceShareType = exports.ColumnStatisticsType = exports.BlueprintRunState = exports.ConditionCheckFailureException = exports.RegistryStatus = exports.ConflictException = exports.SchedulerTransitioningException = exports.CrawlerRunningException = exports.ResourceType = exports.PrincipalType = exports.SessionStatus = exports.S3EncryptionMode = exports.JobBookmarksEncryptionMode = exports.CloudWatchEncryptionMode = exports.Language = void 0;
exports.DeleteTriggerRequestFilterSensitiveLog = exports.DeleteTableVersionResponseFilterSensitiveLog = exports.DeleteTableVersionRequestFilterSensitiveLog = exports.DeleteTableResponseFilterSensitiveLog = exports.DeleteTableRequestFilterSensitiveLog = exports.DeleteSessionResponseFilterSensitiveLog = exports.DeleteSessionRequestFilterSensitiveLog = exports.DeleteSecurityConfigurationResponseFilterSensitiveLog = exports.DeleteSecurityConfigurationRequestFilterSensitiveLog = exports.DeleteSchemaVersionsResponseFilterSensitiveLog = exports.SchemaVersionErrorItemFilterSensitiveLog = exports.ErrorDetailsFilterSensitiveLog = exports.DeleteSchemaVersionsInputFilterSensitiveLog = exports.DeleteSchemaResponseFilterSensitiveLog = exports.DeleteSchemaInputFilterSensitiveLog = exports.DeleteResourcePolicyResponseFilterSensitiveLog = exports.DeleteResourcePolicyRequestFilterSensitiveLog = exports.DeleteRegistryResponseFilterSensitiveLog = exports.DeleteRegistryInputFilterSensitiveLog = exports.DeletePartitionIndexResponseFilterSensitiveLog = exports.DeletePartitionIndexRequestFilterSensitiveLog = exports.DeletePartitionResponseFilterSensitiveLog = exports.DeletePartitionRequestFilterSensitiveLog = exports.DeleteMLTransformResponseFilterSensitiveLog = exports.DeleteMLTransformRequestFilterSensitiveLog = exports.DeleteJobResponseFilterSensitiveLog = exports.DeleteJobRequestFilterSensitiveLog = exports.DeleteDevEndpointResponseFilterSensitiveLog = exports.DeleteDevEndpointRequestFilterSensitiveLog = exports.DeleteDataQualityRulesetResponseFilterSensitiveLog = exports.DeleteDataQualityRulesetRequestFilterSensitiveLog = exports.DeleteDatabaseResponseFilterSensitiveLog = exports.DeleteDatabaseRequestFilterSensitiveLog = exports.DeleteCustomEntityTypeResponseFilterSensitiveLog = exports.DeleteCustomEntityTypeRequestFilterSensitiveLog = exports.DeleteCrawlerResponseFilterSensitiveLog = exports.DeleteCrawlerRequestFilterSensitiveLog = exports.DeleteConnectionResponseFilterSensitiveLog = exports.DeleteConnectionRequestFilterSensitiveLog = exports.DeleteColumnStatisticsForTableResponseFilterSensitiveLog = exports.DeleteColumnStatisticsForTableRequestFilterSensitiveLog = exports.DeleteColumnStatisticsForPartitionResponseFilterSensitiveLog = exports.DeleteColumnStatisticsForPartitionRequestFilterSensitiveLog = exports.DeleteClassifierResponseFilterSensitiveLog = exports.DeleteClassifierRequestFilterSensitiveLog = exports.DeleteBlueprintResponseFilterSensitiveLog = exports.DeleteBlueprintRequestFilterSensitiveLog = exports.CreateWorkflowResponseFilterSensitiveLog = exports.CreateWorkflowRequestFilterSensitiveLog = exports.CreateUserDefinedFunctionResponseFilterSensitiveLog = void 0;
exports.GetCrawlerMetricsResponseFilterSensitiveLog = exports.CrawlerMetricsFilterSensitiveLog = exports.GetCrawlerMetricsRequestFilterSensitiveLog = exports.GetCrawlerResponseFilterSensitiveLog = exports.GetCrawlerRequestFilterSensitiveLog = exports.GetConnectionsResponseFilterSensitiveLog = exports.GetConnectionsRequestFilterSensitiveLog = exports.GetConnectionsFilterFilterSensitiveLog = exports.GetConnectionResponseFilterSensitiveLog = exports.ConnectionFilterSensitiveLog = exports.GetConnectionRequestFilterSensitiveLog = exports.GetColumnStatisticsForTableResponseFilterSensitiveLog = exports.GetColumnStatisticsForTableRequestFilterSensitiveLog = exports.GetColumnStatisticsForPartitionResponseFilterSensitiveLog = exports.ColumnErrorFilterSensitiveLog = exports.ColumnStatisticsFilterSensitiveLog = exports.ColumnStatisticsDataFilterSensitiveLog = exports.StringColumnStatisticsDataFilterSensitiveLog = exports.LongColumnStatisticsDataFilterSensitiveLog = exports.DoubleColumnStatisticsDataFilterSensitiveLog = exports.DecimalColumnStatisticsDataFilterSensitiveLog = exports.DecimalNumberFilterSensitiveLog = exports.DateColumnStatisticsDataFilterSensitiveLog = exports.BooleanColumnStatisticsDataFilterSensitiveLog = exports.BinaryColumnStatisticsDataFilterSensitiveLog = exports.GetColumnStatisticsForPartitionRequestFilterSensitiveLog = exports.GetClassifiersResponseFilterSensitiveLog = exports.GetClassifiersRequestFilterSensitiveLog = exports.GetClassifierResponseFilterSensitiveLog = exports.ClassifierFilterSensitiveLog = exports.XMLClassifierFilterSensitiveLog = exports.JsonClassifierFilterSensitiveLog = exports.GrokClassifierFilterSensitiveLog = exports.CsvClassifierFilterSensitiveLog = exports.GetClassifierRequestFilterSensitiveLog = exports.GetCatalogImportStatusResponseFilterSensitiveLog = exports.CatalogImportStatusFilterSensitiveLog = exports.GetCatalogImportStatusRequestFilterSensitiveLog = exports.GetBlueprintRunsResponseFilterSensitiveLog = exports.GetBlueprintRunsRequestFilterSensitiveLog = exports.GetBlueprintRunResponseFilterSensitiveLog = exports.BlueprintRunFilterSensitiveLog = exports.GetBlueprintRunRequestFilterSensitiveLog = exports.GetBlueprintResponseFilterSensitiveLog = exports.GetBlueprintRequestFilterSensitiveLog = exports.DeleteWorkflowResponseFilterSensitiveLog = exports.DeleteWorkflowRequestFilterSensitiveLog = exports.DeleteUserDefinedFunctionResponseFilterSensitiveLog = exports.DeleteUserDefinedFunctionRequestFilterSensitiveLog = exports.DeleteTriggerResponseFilterSensitiveLog = void 0;
exports.GetMLTaskRunResponseFilterSensitiveLog = exports.TaskRunPropertiesFilterSensitiveLog = exports.LabelingSetGenerationTaskRunPropertiesFilterSensitiveLog = exports.ImportLabelsTaskRunPropertiesFilterSensitiveLog = exports.FindMatchesTaskRunPropertiesFilterSensitiveLog = exports.ExportLabelsTaskRunPropertiesFilterSensitiveLog = exports.GetMLTaskRunRequestFilterSensitiveLog = exports.GetMappingResponseFilterSensitiveLog = exports.MappingEntryFilterSensitiveLog = exports.GetMappingRequestFilterSensitiveLog = exports.CatalogEntryFilterSensitiveLog = exports.LocationFilterSensitiveLog = exports.GetJobsRequestFilterSensitiveLog = exports.GetJobRunsResponseFilterSensitiveLog = exports.GetJobRunsRequestFilterSensitiveLog = exports.GetJobRunResponseFilterSensitiveLog = exports.GetJobRunRequestFilterSensitiveLog = exports.GetJobBookmarkResponseFilterSensitiveLog = exports.JobBookmarkEntryFilterSensitiveLog = exports.GetJobBookmarkRequestFilterSensitiveLog = exports.GetJobRequestFilterSensitiveLog = exports.GetDevEndpointsResponseFilterSensitiveLog = exports.GetDevEndpointsRequestFilterSensitiveLog = exports.GetDevEndpointResponseFilterSensitiveLog = exports.GetDevEndpointRequestFilterSensitiveLog = exports.GetDataQualityRulesetEvaluationRunResponseFilterSensitiveLog = exports.DataQualityEvaluationRunAdditionalRunOptionsFilterSensitiveLog = exports.GetDataQualityRulesetEvaluationRunRequestFilterSensitiveLog = exports.GetDataQualityRulesetResponseFilterSensitiveLog = exports.GetDataQualityRulesetRequestFilterSensitiveLog = exports.GetDataQualityRuleRecommendationRunResponseFilterSensitiveLog = exports.GetDataQualityRuleRecommendationRunRequestFilterSensitiveLog = exports.GetDataQualityResultResponseFilterSensitiveLog = exports.GetDataQualityResultRequestFilterSensitiveLog = exports.GetDataflowGraphResponseFilterSensitiveLog = exports.GetDataflowGraphRequestFilterSensitiveLog = exports.GetDataCatalogEncryptionSettingsResponseFilterSensitiveLog = exports.DataCatalogEncryptionSettingsFilterSensitiveLog = exports.EncryptionAtRestFilterSensitiveLog = exports.ConnectionPasswordEncryptionFilterSensitiveLog = exports.GetDataCatalogEncryptionSettingsRequestFilterSensitiveLog = exports.GetDatabasesResponseFilterSensitiveLog = exports.GetDatabasesRequestFilterSensitiveLog = exports.GetDatabaseResponseFilterSensitiveLog = exports.DatabaseFilterSensitiveLog = exports.GetDatabaseRequestFilterSensitiveLog = exports.GetCustomEntityTypeResponseFilterSensitiveLog = exports.GetCustomEntityTypeRequestFilterSensitiveLog = exports.GetCrawlersResponseFilterSensitiveLog = exports.GetCrawlersRequestFilterSensitiveLog = void 0;
exports.GetSecurityConfigurationsResponseFilterSensitiveLog = exports.GetSecurityConfigurationsRequestFilterSensitiveLog = exports.GetSecurityConfigurationResponseFilterSensitiveLog = exports.SecurityConfigurationFilterSensitiveLog = exports.GetSecurityConfigurationRequestFilterSensitiveLog = exports.GetSchemaVersionsDiffResponseFilterSensitiveLog = exports.GetSchemaVersionsDiffInputFilterSensitiveLog = exports.GetSchemaVersionResponseFilterSensitiveLog = exports.GetSchemaVersionInputFilterSensitiveLog = exports.SchemaVersionNumberFilterSensitiveLog = exports.GetSchemaByDefinitionResponseFilterSensitiveLog = exports.GetSchemaByDefinitionInputFilterSensitiveLog = exports.GetSchemaResponseFilterSensitiveLog = exports.GetSchemaInputFilterSensitiveLog = exports.GetResourcePolicyResponseFilterSensitiveLog = exports.GetResourcePolicyRequestFilterSensitiveLog = exports.GetResourcePoliciesResponseFilterSensitiveLog = exports.GluePolicyFilterSensitiveLog = exports.GetResourcePoliciesRequestFilterSensitiveLog = exports.GetRegistryResponseFilterSensitiveLog = exports.GetRegistryInputFilterSensitiveLog = exports.GetPlanResponseFilterSensitiveLog = exports.GetPlanRequestFilterSensitiveLog = exports.GetPartitionsResponseFilterSensitiveLog = exports.GetPartitionsRequestFilterSensitiveLog = exports.SegmentFilterSensitiveLog = exports.GetPartitionIndexesResponseFilterSensitiveLog = exports.PartitionIndexDescriptorFilterSensitiveLog = exports.KeySchemaElementFilterSensitiveLog = exports.BackfillErrorFilterSensitiveLog = exports.GetPartitionIndexesRequestFilterSensitiveLog = exports.GetPartitionResponseFilterSensitiveLog = exports.GetPartitionRequestFilterSensitiveLog = exports.GetMLTransformsResponseFilterSensitiveLog = exports.MLTransformFilterSensitiveLog = exports.GetMLTransformsRequestFilterSensitiveLog = exports.TransformSortCriteriaFilterSensitiveLog = exports.TransformFilterCriteriaFilterSensitiveLog = exports.GetMLTransformResponseFilterSensitiveLog = exports.SchemaColumnFilterSensitiveLog = exports.EvaluationMetricsFilterSensitiveLog = exports.FindMatchesMetricsFilterSensitiveLog = exports.ConfusionMatrixFilterSensitiveLog = exports.ColumnImportanceFilterSensitiveLog = exports.GetMLTransformRequestFilterSensitiveLog = exports.GetMLTaskRunsResponseFilterSensitiveLog = exports.TaskRunFilterSensitiveLog = exports.GetMLTaskRunsRequestFilterSensitiveLog = exports.TaskRunSortCriteriaFilterSensitiveLog = exports.TaskRunFilterCriteriaFilterSensitiveLog = void 0;
exports.ListCrawlersResponseFilterSensitiveLog = exports.ListCrawlersRequestFilterSensitiveLog = exports.ListBlueprintsResponseFilterSensitiveLog = exports.ListBlueprintsRequestFilterSensitiveLog = exports.ImportCatalogToGlueResponseFilterSensitiveLog = exports.ImportCatalogToGlueRequestFilterSensitiveLog = exports.GetWorkflowRunsResponseFilterSensitiveLog = exports.GetWorkflowRunsRequestFilterSensitiveLog = exports.GetWorkflowRunPropertiesResponseFilterSensitiveLog = exports.GetWorkflowRunPropertiesRequestFilterSensitiveLog = exports.GetWorkflowRunResponseFilterSensitiveLog = exports.GetWorkflowRunRequestFilterSensitiveLog = exports.GetWorkflowResponseFilterSensitiveLog = exports.GetWorkflowRequestFilterSensitiveLog = exports.GetUserDefinedFunctionsResponseFilterSensitiveLog = exports.GetUserDefinedFunctionsRequestFilterSensitiveLog = exports.GetUserDefinedFunctionResponseFilterSensitiveLog = exports.UserDefinedFunctionFilterSensitiveLog = exports.GetUserDefinedFunctionRequestFilterSensitiveLog = exports.GetUnfilteredTableMetadataResponseFilterSensitiveLog = exports.ColumnRowFilterFilterSensitiveLog = exports.GetUnfilteredTableMetadataRequestFilterSensitiveLog = exports.GetUnfilteredPartitionsMetadataResponseFilterSensitiveLog = exports.UnfilteredPartitionFilterSensitiveLog = exports.GetUnfilteredPartitionsMetadataRequestFilterSensitiveLog = exports.GetUnfilteredPartitionMetadataResponseFilterSensitiveLog = exports.GetUnfilteredPartitionMetadataRequestFilterSensitiveLog = exports.GetTriggersResponseFilterSensitiveLog = exports.GetTriggersRequestFilterSensitiveLog = exports.GetTriggerResponseFilterSensitiveLog = exports.GetTriggerRequestFilterSensitiveLog = exports.GetTagsResponseFilterSensitiveLog = exports.GetTagsRequestFilterSensitiveLog = exports.GetTableVersionsResponseFilterSensitiveLog = exports.GetTableVersionsRequestFilterSensitiveLog = exports.GetTableVersionResponseFilterSensitiveLog = exports.TableVersionFilterSensitiveLog = exports.GetTableVersionRequestFilterSensitiveLog = exports.GetTablesResponseFilterSensitiveLog = exports.GetTablesRequestFilterSensitiveLog = exports.GetTableResponseFilterSensitiveLog = exports.TableFilterSensitiveLog = exports.GetTableRequestFilterSensitiveLog = exports.GetStatementResponseFilterSensitiveLog = exports.StatementFilterSensitiveLog = exports.StatementOutputFilterSensitiveLog = exports.StatementOutputDataFilterSensitiveLog = exports.GetStatementRequestFilterSensitiveLog = exports.GetSessionResponseFilterSensitiveLog = exports.GetSessionRequestFilterSensitiveLog = void 0;
const GlueServiceException_1 = require("./GlueServiceException");
var Language;
(function (Language) {
    Language["PYTHON"] = "PYTHON";
    Language["SCALA"] = "SCALA";
})(Language = exports.Language || (exports.Language = {}));
var CloudWatchEncryptionMode;
(function (CloudWatchEncryptionMode) {
    CloudWatchEncryptionMode["DISABLED"] = "DISABLED";
    CloudWatchEncryptionMode["SSEKMS"] = "SSE-KMS";
})(CloudWatchEncryptionMode = exports.CloudWatchEncryptionMode || (exports.CloudWatchEncryptionMode = {}));
var JobBookmarksEncryptionMode;
(function (JobBookmarksEncryptionMode) {
    JobBookmarksEncryptionMode["CSEKMS"] = "CSE-KMS";
    JobBookmarksEncryptionMode["DISABLED"] = "DISABLED";
})(JobBookmarksEncryptionMode = exports.JobBookmarksEncryptionMode || (exports.JobBookmarksEncryptionMode = {}));
var S3EncryptionMode;
(function (S3EncryptionMode) {
    S3EncryptionMode["DISABLED"] = "DISABLED";
    S3EncryptionMode["SSEKMS"] = "SSE-KMS";
    S3EncryptionMode["SSES3"] = "SSE-S3";
})(S3EncryptionMode = exports.S3EncryptionMode || (exports.S3EncryptionMode = {}));
var SessionStatus;
(function (SessionStatus) {
    SessionStatus["FAILED"] = "FAILED";
    SessionStatus["PROVISIONING"] = "PROVISIONING";
    SessionStatus["READY"] = "READY";
    SessionStatus["STOPPED"] = "STOPPED";
    SessionStatus["STOPPING"] = "STOPPING";
    SessionStatus["TIMEOUT"] = "TIMEOUT";
})(SessionStatus = exports.SessionStatus || (exports.SessionStatus = {}));
var PrincipalType;
(function (PrincipalType) {
    PrincipalType["GROUP"] = "GROUP";
    PrincipalType["ROLE"] = "ROLE";
    PrincipalType["USER"] = "USER";
})(PrincipalType = exports.PrincipalType || (exports.PrincipalType = {}));
var ResourceType;
(function (ResourceType) {
    ResourceType["ARCHIVE"] = "ARCHIVE";
    ResourceType["FILE"] = "FILE";
    ResourceType["JAR"] = "JAR";
})(ResourceType = exports.ResourceType || (exports.ResourceType = {}));
class CrawlerRunningException extends GlueServiceException_1.GlueServiceException {
    constructor(opts) {
        super({
            name: "CrawlerRunningException",
            $fault: "client",
            ...opts,
        });
        this.name = "CrawlerRunningException";
        this.$fault = "client";
        Object.setPrototypeOf(this, CrawlerRunningException.prototype);
        this.Message = opts.Message;
    }
}
exports.CrawlerRunningException = CrawlerRunningException;
class SchedulerTransitioningException extends GlueServiceException_1.GlueServiceException {
    constructor(opts) {
        super({
            name: "SchedulerTransitioningException",
            $fault: "client",
            ...opts,
        });
        this.name = "SchedulerTransitioningException";
        this.$fault = "client";
        Object.setPrototypeOf(this, SchedulerTransitioningException.prototype);
        this.Message = opts.Message;
    }
}
exports.SchedulerTransitioningException = SchedulerTransitioningException;
class ConflictException extends GlueServiceException_1.GlueServiceException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
exports.ConflictException = ConflictException;
var RegistryStatus;
(function (RegistryStatus) {
    RegistryStatus["AVAILABLE"] = "AVAILABLE";
    RegistryStatus["DELETING"] = "DELETING";
})(RegistryStatus = exports.RegistryStatus || (exports.RegistryStatus = {}));
class ConditionCheckFailureException extends GlueServiceException_1.GlueServiceException {
    constructor(opts) {
        super({
            name: "ConditionCheckFailureException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConditionCheckFailureException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConditionCheckFailureException.prototype);
        this.Message = opts.Message;
    }
}
exports.ConditionCheckFailureException = ConditionCheckFailureException;
var BlueprintRunState;
(function (BlueprintRunState) {
    BlueprintRunState["FAILED"] = "FAILED";
    BlueprintRunState["ROLLING_BACK"] = "ROLLING_BACK";
    BlueprintRunState["RUNNING"] = "RUNNING";
    BlueprintRunState["SUCCEEDED"] = "SUCCEEDED";
})(BlueprintRunState = exports.BlueprintRunState || (exports.BlueprintRunState = {}));
var ColumnStatisticsType;
(function (ColumnStatisticsType) {
    ColumnStatisticsType["BINARY"] = "BINARY";
    ColumnStatisticsType["BOOLEAN"] = "BOOLEAN";
    ColumnStatisticsType["DATE"] = "DATE";
    ColumnStatisticsType["DECIMAL"] = "DECIMAL";
    ColumnStatisticsType["DOUBLE"] = "DOUBLE";
    ColumnStatisticsType["LONG"] = "LONG";
    ColumnStatisticsType["STRING"] = "STRING";
})(ColumnStatisticsType = exports.ColumnStatisticsType || (exports.ColumnStatisticsType = {}));
var ResourceShareType;
(function (ResourceShareType) {
    ResourceShareType["ALL"] = "ALL";
    ResourceShareType["FOREIGN"] = "FOREIGN";
})(ResourceShareType = exports.ResourceShareType || (exports.ResourceShareType = {}));
var CatalogEncryptionMode;
(function (CatalogEncryptionMode) {
    CatalogEncryptionMode["DISABLED"] = "DISABLED";
    CatalogEncryptionMode["SSEKMS"] = "SSE-KMS";
})(CatalogEncryptionMode = exports.CatalogEncryptionMode || (exports.CatalogEncryptionMode = {}));
var TaskType;
(function (TaskType) {
    TaskType["EVALUATION"] = "EVALUATION";
    TaskType["EXPORT_LABELS"] = "EXPORT_LABELS";
    TaskType["FIND_MATCHES"] = "FIND_MATCHES";
    TaskType["IMPORT_LABELS"] = "IMPORT_LABELS";
    TaskType["LABELING_SET_GENERATION"] = "LABELING_SET_GENERATION";
})(TaskType = exports.TaskType || (exports.TaskType = {}));
var TaskRunSortColumnType;
(function (TaskRunSortColumnType) {
    TaskRunSortColumnType["STARTED"] = "STARTED";
    TaskRunSortColumnType["STATUS"] = "STATUS";
    TaskRunSortColumnType["TASK_RUN_TYPE"] = "TASK_RUN_TYPE";
})(TaskRunSortColumnType = exports.TaskRunSortColumnType || (exports.TaskRunSortColumnType = {}));
var SortDirectionType;
(function (SortDirectionType) {
    SortDirectionType["ASCENDING"] = "ASCENDING";
    SortDirectionType["DESCENDING"] = "DESCENDING";
})(SortDirectionType = exports.SortDirectionType || (exports.SortDirectionType = {}));
var TransformStatusType;
(function (TransformStatusType) {
    TransformStatusType["DELETING"] = "DELETING";
    TransformStatusType["NOT_READY"] = "NOT_READY";
    TransformStatusType["READY"] = "READY";
})(TransformStatusType = exports.TransformStatusType || (exports.TransformStatusType = {}));
var TransformSortColumnType;
(function (TransformSortColumnType) {
    TransformSortColumnType["CREATED"] = "CREATED";
    TransformSortColumnType["LAST_MODIFIED"] = "LAST_MODIFIED";
    TransformSortColumnType["NAME"] = "NAME";
    TransformSortColumnType["STATUS"] = "STATUS";
    TransformSortColumnType["TRANSFORM_TYPE"] = "TRANSFORM_TYPE";
})(TransformSortColumnType = exports.TransformSortColumnType || (exports.TransformSortColumnType = {}));
var BackfillErrorCode;
(function (BackfillErrorCode) {
    BackfillErrorCode["ENCRYPTED_PARTITION_ERROR"] = "ENCRYPTED_PARTITION_ERROR";
    BackfillErrorCode["INTERNAL_ERROR"] = "INTERNAL_ERROR";
    BackfillErrorCode["INVALID_PARTITION_TYPE_DATA_ERROR"] = "INVALID_PARTITION_TYPE_DATA_ERROR";
    BackfillErrorCode["MISSING_PARTITION_VALUE_ERROR"] = "MISSING_PARTITION_VALUE_ERROR";
    BackfillErrorCode["UNSUPPORTED_PARTITION_CHARACTER_ERROR"] = "UNSUPPORTED_PARTITION_CHARACTER_ERROR";
})(BackfillErrorCode = exports.BackfillErrorCode || (exports.BackfillErrorCode = {}));
var PartitionIndexStatus;
(function (PartitionIndexStatus) {
    PartitionIndexStatus["ACTIVE"] = "ACTIVE";
    PartitionIndexStatus["CREATING"] = "CREATING";
    PartitionIndexStatus["DELETING"] = "DELETING";
    PartitionIndexStatus["FAILED"] = "FAILED";
})(PartitionIndexStatus = exports.PartitionIndexStatus || (exports.PartitionIndexStatus = {}));
var SchemaDiffType;
(function (SchemaDiffType) {
    SchemaDiffType["SYNTAX_DIFF"] = "SYNTAX_DIFF";
})(SchemaDiffType = exports.SchemaDiffType || (exports.SchemaDiffType = {}));
var StatementState;
(function (StatementState) {
    StatementState["AVAILABLE"] = "AVAILABLE";
    StatementState["CANCELLED"] = "CANCELLED";
    StatementState["CANCELLING"] = "CANCELLING";
    StatementState["ERROR"] = "ERROR";
    StatementState["RUNNING"] = "RUNNING";
    StatementState["WAITING"] = "WAITING";
})(StatementState = exports.StatementState || (exports.StatementState = {}));
var PermissionType;
(function (PermissionType) {
    PermissionType["CELL_FILTER_PERMISSION"] = "CELL_FILTER_PERMISSION";
    PermissionType["COLUMN_PERMISSION"] = "COLUMN_PERMISSION";
})(PermissionType = exports.PermissionType || (exports.PermissionType = {}));
class PermissionTypeMismatchException extends GlueServiceException_1.GlueServiceException {
    constructor(opts) {
        super({
            name: "PermissionTypeMismatchException",
            $fault: "client",
            ...opts,
        });
        this.name = "PermissionTypeMismatchException";
        this.$fault = "client";
        Object.setPrototypeOf(this, PermissionTypeMismatchException.prototype);
        this.Message = opts.Message;
    }
}
exports.PermissionTypeMismatchException = PermissionTypeMismatchException;
var FieldName;
(function (FieldName) {
    FieldName["CRAWL_ID"] = "CRAWL_ID";
    FieldName["DPU_HOUR"] = "DPU_HOUR";
    FieldName["END_TIME"] = "END_TIME";
    FieldName["START_TIME"] = "START_TIME";
    FieldName["STATE"] = "STATE";
})(FieldName = exports.FieldName || (exports.FieldName = {}));
var FilterOperator;
(function (FilterOperator) {
    FilterOperator["EQ"] = "EQ";
    FilterOperator["GE"] = "GE";
    FilterOperator["GT"] = "GT";
    FilterOperator["LE"] = "LE";
    FilterOperator["LT"] = "LT";
    FilterOperator["NE"] = "NE";
})(FilterOperator = exports.FilterOperator || (exports.FilterOperator = {}));
const CreateScriptRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateScriptRequestFilterSensitiveLog = CreateScriptRequestFilterSensitiveLog;
const CreateScriptResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateScriptResponseFilterSensitiveLog = CreateScriptResponseFilterSensitiveLog;
const CloudWatchEncryptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CloudWatchEncryptionFilterSensitiveLog = CloudWatchEncryptionFilterSensitiveLog;
const JobBookmarksEncryptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.JobBookmarksEncryptionFilterSensitiveLog = JobBookmarksEncryptionFilterSensitiveLog;
const S3EncryptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.S3EncryptionFilterSensitiveLog = S3EncryptionFilterSensitiveLog;
const EncryptionConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EncryptionConfigurationFilterSensitiveLog = EncryptionConfigurationFilterSensitiveLog;
const CreateSecurityConfigurationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateSecurityConfigurationRequestFilterSensitiveLog = CreateSecurityConfigurationRequestFilterSensitiveLog;
const CreateSecurityConfigurationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateSecurityConfigurationResponseFilterSensitiveLog = CreateSecurityConfigurationResponseFilterSensitiveLog;
const SessionCommandFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SessionCommandFilterSensitiveLog = SessionCommandFilterSensitiveLog;
const CreateSessionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateSessionRequestFilterSensitiveLog = CreateSessionRequestFilterSensitiveLog;
const SessionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SessionFilterSensitiveLog = SessionFilterSensitiveLog;
const CreateSessionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateSessionResponseFilterSensitiveLog = CreateSessionResponseFilterSensitiveLog;
const TableIdentifierFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TableIdentifierFilterSensitiveLog = TableIdentifierFilterSensitiveLog;
const TableInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TableInputFilterSensitiveLog = TableInputFilterSensitiveLog;
const CreateTableRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateTableRequestFilterSensitiveLog = CreateTableRequestFilterSensitiveLog;
const CreateTableResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateTableResponseFilterSensitiveLog = CreateTableResponseFilterSensitiveLog;
const CreateTriggerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateTriggerRequestFilterSensitiveLog = CreateTriggerRequestFilterSensitiveLog;
const CreateTriggerResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateTriggerResponseFilterSensitiveLog = CreateTriggerResponseFilterSensitiveLog;
const ResourceUriFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResourceUriFilterSensitiveLog = ResourceUriFilterSensitiveLog;
const UserDefinedFunctionInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UserDefinedFunctionInputFilterSensitiveLog = UserDefinedFunctionInputFilterSensitiveLog;
const CreateUserDefinedFunctionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateUserDefinedFunctionRequestFilterSensitiveLog = CreateUserDefinedFunctionRequestFilterSensitiveLog;
const CreateUserDefinedFunctionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateUserDefinedFunctionResponseFilterSensitiveLog = CreateUserDefinedFunctionResponseFilterSensitiveLog;
const CreateWorkflowRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateWorkflowRequestFilterSensitiveLog = CreateWorkflowRequestFilterSensitiveLog;
const CreateWorkflowResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateWorkflowResponseFilterSensitiveLog = CreateWorkflowResponseFilterSensitiveLog;
const DeleteBlueprintRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteBlueprintRequestFilterSensitiveLog = DeleteBlueprintRequestFilterSensitiveLog;
const DeleteBlueprintResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteBlueprintResponseFilterSensitiveLog = DeleteBlueprintResponseFilterSensitiveLog;
const DeleteClassifierRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteClassifierRequestFilterSensitiveLog = DeleteClassifierRequestFilterSensitiveLog;
const DeleteClassifierResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteClassifierResponseFilterSensitiveLog = DeleteClassifierResponseFilterSensitiveLog;
const DeleteColumnStatisticsForPartitionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteColumnStatisticsForPartitionRequestFilterSensitiveLog = DeleteColumnStatisticsForPartitionRequestFilterSensitiveLog;
const DeleteColumnStatisticsForPartitionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteColumnStatisticsForPartitionResponseFilterSensitiveLog = DeleteColumnStatisticsForPartitionResponseFilterSensitiveLog;
const DeleteColumnStatisticsForTableRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteColumnStatisticsForTableRequestFilterSensitiveLog = DeleteColumnStatisticsForTableRequestFilterSensitiveLog;
const DeleteColumnStatisticsForTableResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteColumnStatisticsForTableResponseFilterSensitiveLog = DeleteColumnStatisticsForTableResponseFilterSensitiveLog;
const DeleteConnectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteConnectionRequestFilterSensitiveLog = DeleteConnectionRequestFilterSensitiveLog;
const DeleteConnectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteConnectionResponseFilterSensitiveLog = DeleteConnectionResponseFilterSensitiveLog;
const DeleteCrawlerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteCrawlerRequestFilterSensitiveLog = DeleteCrawlerRequestFilterSensitiveLog;
const DeleteCrawlerResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteCrawlerResponseFilterSensitiveLog = DeleteCrawlerResponseFilterSensitiveLog;
const DeleteCustomEntityTypeRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteCustomEntityTypeRequestFilterSensitiveLog = DeleteCustomEntityTypeRequestFilterSensitiveLog;
const DeleteCustomEntityTypeResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteCustomEntityTypeResponseFilterSensitiveLog = DeleteCustomEntityTypeResponseFilterSensitiveLog;
const DeleteDatabaseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDatabaseRequestFilterSensitiveLog = DeleteDatabaseRequestFilterSensitiveLog;
const DeleteDatabaseResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDatabaseResponseFilterSensitiveLog = DeleteDatabaseResponseFilterSensitiveLog;
const DeleteDataQualityRulesetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDataQualityRulesetRequestFilterSensitiveLog = DeleteDataQualityRulesetRequestFilterSensitiveLog;
const DeleteDataQualityRulesetResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDataQualityRulesetResponseFilterSensitiveLog = DeleteDataQualityRulesetResponseFilterSensitiveLog;
const DeleteDevEndpointRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDevEndpointRequestFilterSensitiveLog = DeleteDevEndpointRequestFilterSensitiveLog;
const DeleteDevEndpointResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDevEndpointResponseFilterSensitiveLog = DeleteDevEndpointResponseFilterSensitiveLog;
const DeleteJobRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteJobRequestFilterSensitiveLog = DeleteJobRequestFilterSensitiveLog;
const DeleteJobResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteJobResponseFilterSensitiveLog = DeleteJobResponseFilterSensitiveLog;
const DeleteMLTransformRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteMLTransformRequestFilterSensitiveLog = DeleteMLTransformRequestFilterSensitiveLog;
const DeleteMLTransformResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteMLTransformResponseFilterSensitiveLog = DeleteMLTransformResponseFilterSensitiveLog;
const DeletePartitionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeletePartitionRequestFilterSensitiveLog = DeletePartitionRequestFilterSensitiveLog;
const DeletePartitionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeletePartitionResponseFilterSensitiveLog = DeletePartitionResponseFilterSensitiveLog;
const DeletePartitionIndexRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeletePartitionIndexRequestFilterSensitiveLog = DeletePartitionIndexRequestFilterSensitiveLog;
const DeletePartitionIndexResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeletePartitionIndexResponseFilterSensitiveLog = DeletePartitionIndexResponseFilterSensitiveLog;
const DeleteRegistryInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteRegistryInputFilterSensitiveLog = DeleteRegistryInputFilterSensitiveLog;
const DeleteRegistryResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteRegistryResponseFilterSensitiveLog = DeleteRegistryResponseFilterSensitiveLog;
const DeleteResourcePolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteResourcePolicyRequestFilterSensitiveLog = DeleteResourcePolicyRequestFilterSensitiveLog;
const DeleteResourcePolicyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteResourcePolicyResponseFilterSensitiveLog = DeleteResourcePolicyResponseFilterSensitiveLog;
const DeleteSchemaInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSchemaInputFilterSensitiveLog = DeleteSchemaInputFilterSensitiveLog;
const DeleteSchemaResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSchemaResponseFilterSensitiveLog = DeleteSchemaResponseFilterSensitiveLog;
const DeleteSchemaVersionsInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSchemaVersionsInputFilterSensitiveLog = DeleteSchemaVersionsInputFilterSensitiveLog;
const ErrorDetailsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ErrorDetailsFilterSensitiveLog = ErrorDetailsFilterSensitiveLog;
const SchemaVersionErrorItemFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SchemaVersionErrorItemFilterSensitiveLog = SchemaVersionErrorItemFilterSensitiveLog;
const DeleteSchemaVersionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSchemaVersionsResponseFilterSensitiveLog = DeleteSchemaVersionsResponseFilterSensitiveLog;
const DeleteSecurityConfigurationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSecurityConfigurationRequestFilterSensitiveLog = DeleteSecurityConfigurationRequestFilterSensitiveLog;
const DeleteSecurityConfigurationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSecurityConfigurationResponseFilterSensitiveLog = DeleteSecurityConfigurationResponseFilterSensitiveLog;
const DeleteSessionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSessionRequestFilterSensitiveLog = DeleteSessionRequestFilterSensitiveLog;
const DeleteSessionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSessionResponseFilterSensitiveLog = DeleteSessionResponseFilterSensitiveLog;
const DeleteTableRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteTableRequestFilterSensitiveLog = DeleteTableRequestFilterSensitiveLog;
const DeleteTableResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteTableResponseFilterSensitiveLog = DeleteTableResponseFilterSensitiveLog;
const DeleteTableVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteTableVersionRequestFilterSensitiveLog = DeleteTableVersionRequestFilterSensitiveLog;
const DeleteTableVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteTableVersionResponseFilterSensitiveLog = DeleteTableVersionResponseFilterSensitiveLog;
const DeleteTriggerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteTriggerRequestFilterSensitiveLog = DeleteTriggerRequestFilterSensitiveLog;
const DeleteTriggerResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteTriggerResponseFilterSensitiveLog = DeleteTriggerResponseFilterSensitiveLog;
const DeleteUserDefinedFunctionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteUserDefinedFunctionRequestFilterSensitiveLog = DeleteUserDefinedFunctionRequestFilterSensitiveLog;
const DeleteUserDefinedFunctionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteUserDefinedFunctionResponseFilterSensitiveLog = DeleteUserDefinedFunctionResponseFilterSensitiveLog;
const DeleteWorkflowRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteWorkflowRequestFilterSensitiveLog = DeleteWorkflowRequestFilterSensitiveLog;
const DeleteWorkflowResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteWorkflowResponseFilterSensitiveLog = DeleteWorkflowResponseFilterSensitiveLog;
const GetBlueprintRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBlueprintRequestFilterSensitiveLog = GetBlueprintRequestFilterSensitiveLog;
const GetBlueprintResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBlueprintResponseFilterSensitiveLog = GetBlueprintResponseFilterSensitiveLog;
const GetBlueprintRunRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBlueprintRunRequestFilterSensitiveLog = GetBlueprintRunRequestFilterSensitiveLog;
const BlueprintRunFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BlueprintRunFilterSensitiveLog = BlueprintRunFilterSensitiveLog;
const GetBlueprintRunResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBlueprintRunResponseFilterSensitiveLog = GetBlueprintRunResponseFilterSensitiveLog;
const GetBlueprintRunsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBlueprintRunsRequestFilterSensitiveLog = GetBlueprintRunsRequestFilterSensitiveLog;
const GetBlueprintRunsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBlueprintRunsResponseFilterSensitiveLog = GetBlueprintRunsResponseFilterSensitiveLog;
const GetCatalogImportStatusRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCatalogImportStatusRequestFilterSensitiveLog = GetCatalogImportStatusRequestFilterSensitiveLog;
const CatalogImportStatusFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CatalogImportStatusFilterSensitiveLog = CatalogImportStatusFilterSensitiveLog;
const GetCatalogImportStatusResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCatalogImportStatusResponseFilterSensitiveLog = GetCatalogImportStatusResponseFilterSensitiveLog;
const GetClassifierRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetClassifierRequestFilterSensitiveLog = GetClassifierRequestFilterSensitiveLog;
const CsvClassifierFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CsvClassifierFilterSensitiveLog = CsvClassifierFilterSensitiveLog;
const GrokClassifierFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GrokClassifierFilterSensitiveLog = GrokClassifierFilterSensitiveLog;
const JsonClassifierFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.JsonClassifierFilterSensitiveLog = JsonClassifierFilterSensitiveLog;
const XMLClassifierFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.XMLClassifierFilterSensitiveLog = XMLClassifierFilterSensitiveLog;
const ClassifierFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ClassifierFilterSensitiveLog = ClassifierFilterSensitiveLog;
const GetClassifierResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetClassifierResponseFilterSensitiveLog = GetClassifierResponseFilterSensitiveLog;
const GetClassifiersRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetClassifiersRequestFilterSensitiveLog = GetClassifiersRequestFilterSensitiveLog;
const GetClassifiersResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetClassifiersResponseFilterSensitiveLog = GetClassifiersResponseFilterSensitiveLog;
const GetColumnStatisticsForPartitionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetColumnStatisticsForPartitionRequestFilterSensitiveLog = GetColumnStatisticsForPartitionRequestFilterSensitiveLog;
const BinaryColumnStatisticsDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BinaryColumnStatisticsDataFilterSensitiveLog = BinaryColumnStatisticsDataFilterSensitiveLog;
const BooleanColumnStatisticsDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BooleanColumnStatisticsDataFilterSensitiveLog = BooleanColumnStatisticsDataFilterSensitiveLog;
const DateColumnStatisticsDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DateColumnStatisticsDataFilterSensitiveLog = DateColumnStatisticsDataFilterSensitiveLog;
const DecimalNumberFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DecimalNumberFilterSensitiveLog = DecimalNumberFilterSensitiveLog;
const DecimalColumnStatisticsDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DecimalColumnStatisticsDataFilterSensitiveLog = DecimalColumnStatisticsDataFilterSensitiveLog;
const DoubleColumnStatisticsDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DoubleColumnStatisticsDataFilterSensitiveLog = DoubleColumnStatisticsDataFilterSensitiveLog;
const LongColumnStatisticsDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LongColumnStatisticsDataFilterSensitiveLog = LongColumnStatisticsDataFilterSensitiveLog;
const StringColumnStatisticsDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StringColumnStatisticsDataFilterSensitiveLog = StringColumnStatisticsDataFilterSensitiveLog;
const ColumnStatisticsDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ColumnStatisticsDataFilterSensitiveLog = ColumnStatisticsDataFilterSensitiveLog;
const ColumnStatisticsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ColumnStatisticsFilterSensitiveLog = ColumnStatisticsFilterSensitiveLog;
const ColumnErrorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ColumnErrorFilterSensitiveLog = ColumnErrorFilterSensitiveLog;
const GetColumnStatisticsForPartitionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetColumnStatisticsForPartitionResponseFilterSensitiveLog = GetColumnStatisticsForPartitionResponseFilterSensitiveLog;
const GetColumnStatisticsForTableRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetColumnStatisticsForTableRequestFilterSensitiveLog = GetColumnStatisticsForTableRequestFilterSensitiveLog;
const GetColumnStatisticsForTableResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetColumnStatisticsForTableResponseFilterSensitiveLog = GetColumnStatisticsForTableResponseFilterSensitiveLog;
const GetConnectionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConnectionRequestFilterSensitiveLog = GetConnectionRequestFilterSensitiveLog;
const ConnectionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConnectionFilterSensitiveLog = ConnectionFilterSensitiveLog;
const GetConnectionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConnectionResponseFilterSensitiveLog = GetConnectionResponseFilterSensitiveLog;
const GetConnectionsFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConnectionsFilterFilterSensitiveLog = GetConnectionsFilterFilterSensitiveLog;
const GetConnectionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConnectionsRequestFilterSensitiveLog = GetConnectionsRequestFilterSensitiveLog;
const GetConnectionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConnectionsResponseFilterSensitiveLog = GetConnectionsResponseFilterSensitiveLog;
const GetCrawlerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCrawlerRequestFilterSensitiveLog = GetCrawlerRequestFilterSensitiveLog;
const GetCrawlerResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCrawlerResponseFilterSensitiveLog = GetCrawlerResponseFilterSensitiveLog;
const GetCrawlerMetricsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCrawlerMetricsRequestFilterSensitiveLog = GetCrawlerMetricsRequestFilterSensitiveLog;
const CrawlerMetricsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CrawlerMetricsFilterSensitiveLog = CrawlerMetricsFilterSensitiveLog;
const GetCrawlerMetricsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCrawlerMetricsResponseFilterSensitiveLog = GetCrawlerMetricsResponseFilterSensitiveLog;
const GetCrawlersRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCrawlersRequestFilterSensitiveLog = GetCrawlersRequestFilterSensitiveLog;
const GetCrawlersResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCrawlersResponseFilterSensitiveLog = GetCrawlersResponseFilterSensitiveLog;
const GetCustomEntityTypeRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCustomEntityTypeRequestFilterSensitiveLog = GetCustomEntityTypeRequestFilterSensitiveLog;
const GetCustomEntityTypeResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCustomEntityTypeResponseFilterSensitiveLog = GetCustomEntityTypeResponseFilterSensitiveLog;
const GetDatabaseRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDatabaseRequestFilterSensitiveLog = GetDatabaseRequestFilterSensitiveLog;
const DatabaseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatabaseFilterSensitiveLog = DatabaseFilterSensitiveLog;
const GetDatabaseResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDatabaseResponseFilterSensitiveLog = GetDatabaseResponseFilterSensitiveLog;
const GetDatabasesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDatabasesRequestFilterSensitiveLog = GetDatabasesRequestFilterSensitiveLog;
const GetDatabasesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDatabasesResponseFilterSensitiveLog = GetDatabasesResponseFilterSensitiveLog;
const GetDataCatalogEncryptionSettingsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataCatalogEncryptionSettingsRequestFilterSensitiveLog = GetDataCatalogEncryptionSettingsRequestFilterSensitiveLog;
const ConnectionPasswordEncryptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConnectionPasswordEncryptionFilterSensitiveLog = ConnectionPasswordEncryptionFilterSensitiveLog;
const EncryptionAtRestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EncryptionAtRestFilterSensitiveLog = EncryptionAtRestFilterSensitiveLog;
const DataCatalogEncryptionSettingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DataCatalogEncryptionSettingsFilterSensitiveLog = DataCatalogEncryptionSettingsFilterSensitiveLog;
const GetDataCatalogEncryptionSettingsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataCatalogEncryptionSettingsResponseFilterSensitiveLog = GetDataCatalogEncryptionSettingsResponseFilterSensitiveLog;
const GetDataflowGraphRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataflowGraphRequestFilterSensitiveLog = GetDataflowGraphRequestFilterSensitiveLog;
const GetDataflowGraphResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataflowGraphResponseFilterSensitiveLog = GetDataflowGraphResponseFilterSensitiveLog;
const GetDataQualityResultRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataQualityResultRequestFilterSensitiveLog = GetDataQualityResultRequestFilterSensitiveLog;
const GetDataQualityResultResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataQualityResultResponseFilterSensitiveLog = GetDataQualityResultResponseFilterSensitiveLog;
const GetDataQualityRuleRecommendationRunRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataQualityRuleRecommendationRunRequestFilterSensitiveLog = GetDataQualityRuleRecommendationRunRequestFilterSensitiveLog;
const GetDataQualityRuleRecommendationRunResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataQualityRuleRecommendationRunResponseFilterSensitiveLog = GetDataQualityRuleRecommendationRunResponseFilterSensitiveLog;
const GetDataQualityRulesetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataQualityRulesetRequestFilterSensitiveLog = GetDataQualityRulesetRequestFilterSensitiveLog;
const GetDataQualityRulesetResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataQualityRulesetResponseFilterSensitiveLog = GetDataQualityRulesetResponseFilterSensitiveLog;
const GetDataQualityRulesetEvaluationRunRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataQualityRulesetEvaluationRunRequestFilterSensitiveLog = GetDataQualityRulesetEvaluationRunRequestFilterSensitiveLog;
const DataQualityEvaluationRunAdditionalRunOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DataQualityEvaluationRunAdditionalRunOptionsFilterSensitiveLog = DataQualityEvaluationRunAdditionalRunOptionsFilterSensitiveLog;
const GetDataQualityRulesetEvaluationRunResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataQualityRulesetEvaluationRunResponseFilterSensitiveLog = GetDataQualityRulesetEvaluationRunResponseFilterSensitiveLog;
const GetDevEndpointRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDevEndpointRequestFilterSensitiveLog = GetDevEndpointRequestFilterSensitiveLog;
const GetDevEndpointResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDevEndpointResponseFilterSensitiveLog = GetDevEndpointResponseFilterSensitiveLog;
const GetDevEndpointsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDevEndpointsRequestFilterSensitiveLog = GetDevEndpointsRequestFilterSensitiveLog;
const GetDevEndpointsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDevEndpointsResponseFilterSensitiveLog = GetDevEndpointsResponseFilterSensitiveLog;
const GetJobRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetJobRequestFilterSensitiveLog = GetJobRequestFilterSensitiveLog;
const GetJobBookmarkRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetJobBookmarkRequestFilterSensitiveLog = GetJobBookmarkRequestFilterSensitiveLog;
const JobBookmarkEntryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.JobBookmarkEntryFilterSensitiveLog = JobBookmarkEntryFilterSensitiveLog;
const GetJobBookmarkResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetJobBookmarkResponseFilterSensitiveLog = GetJobBookmarkResponseFilterSensitiveLog;
const GetJobRunRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetJobRunRequestFilterSensitiveLog = GetJobRunRequestFilterSensitiveLog;
const GetJobRunResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetJobRunResponseFilterSensitiveLog = GetJobRunResponseFilterSensitiveLog;
const GetJobRunsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetJobRunsRequestFilterSensitiveLog = GetJobRunsRequestFilterSensitiveLog;
const GetJobRunsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetJobRunsResponseFilterSensitiveLog = GetJobRunsResponseFilterSensitiveLog;
const GetJobsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetJobsRequestFilterSensitiveLog = GetJobsRequestFilterSensitiveLog;
const LocationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LocationFilterSensitiveLog = LocationFilterSensitiveLog;
const CatalogEntryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CatalogEntryFilterSensitiveLog = CatalogEntryFilterSensitiveLog;
const GetMappingRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMappingRequestFilterSensitiveLog = GetMappingRequestFilterSensitiveLog;
const MappingEntryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MappingEntryFilterSensitiveLog = MappingEntryFilterSensitiveLog;
const GetMappingResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMappingResponseFilterSensitiveLog = GetMappingResponseFilterSensitiveLog;
const GetMLTaskRunRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMLTaskRunRequestFilterSensitiveLog = GetMLTaskRunRequestFilterSensitiveLog;
const ExportLabelsTaskRunPropertiesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ExportLabelsTaskRunPropertiesFilterSensitiveLog = ExportLabelsTaskRunPropertiesFilterSensitiveLog;
const FindMatchesTaskRunPropertiesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FindMatchesTaskRunPropertiesFilterSensitiveLog = FindMatchesTaskRunPropertiesFilterSensitiveLog;
const ImportLabelsTaskRunPropertiesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportLabelsTaskRunPropertiesFilterSensitiveLog = ImportLabelsTaskRunPropertiesFilterSensitiveLog;
const LabelingSetGenerationTaskRunPropertiesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.LabelingSetGenerationTaskRunPropertiesFilterSensitiveLog = LabelingSetGenerationTaskRunPropertiesFilterSensitiveLog;
const TaskRunPropertiesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TaskRunPropertiesFilterSensitiveLog = TaskRunPropertiesFilterSensitiveLog;
const GetMLTaskRunResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMLTaskRunResponseFilterSensitiveLog = GetMLTaskRunResponseFilterSensitiveLog;
const TaskRunFilterCriteriaFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TaskRunFilterCriteriaFilterSensitiveLog = TaskRunFilterCriteriaFilterSensitiveLog;
const TaskRunSortCriteriaFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TaskRunSortCriteriaFilterSensitiveLog = TaskRunSortCriteriaFilterSensitiveLog;
const GetMLTaskRunsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMLTaskRunsRequestFilterSensitiveLog = GetMLTaskRunsRequestFilterSensitiveLog;
const TaskRunFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TaskRunFilterSensitiveLog = TaskRunFilterSensitiveLog;
const GetMLTaskRunsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMLTaskRunsResponseFilterSensitiveLog = GetMLTaskRunsResponseFilterSensitiveLog;
const GetMLTransformRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMLTransformRequestFilterSensitiveLog = GetMLTransformRequestFilterSensitiveLog;
const ColumnImportanceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ColumnImportanceFilterSensitiveLog = ColumnImportanceFilterSensitiveLog;
const ConfusionMatrixFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ConfusionMatrixFilterSensitiveLog = ConfusionMatrixFilterSensitiveLog;
const FindMatchesMetricsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FindMatchesMetricsFilterSensitiveLog = FindMatchesMetricsFilterSensitiveLog;
const EvaluationMetricsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EvaluationMetricsFilterSensitiveLog = EvaluationMetricsFilterSensitiveLog;
const SchemaColumnFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SchemaColumnFilterSensitiveLog = SchemaColumnFilterSensitiveLog;
const GetMLTransformResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMLTransformResponseFilterSensitiveLog = GetMLTransformResponseFilterSensitiveLog;
const TransformFilterCriteriaFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TransformFilterCriteriaFilterSensitiveLog = TransformFilterCriteriaFilterSensitiveLog;
const TransformSortCriteriaFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TransformSortCriteriaFilterSensitiveLog = TransformSortCriteriaFilterSensitiveLog;
const GetMLTransformsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMLTransformsRequestFilterSensitiveLog = GetMLTransformsRequestFilterSensitiveLog;
const MLTransformFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MLTransformFilterSensitiveLog = MLTransformFilterSensitiveLog;
const GetMLTransformsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetMLTransformsResponseFilterSensitiveLog = GetMLTransformsResponseFilterSensitiveLog;
const GetPartitionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPartitionRequestFilterSensitiveLog = GetPartitionRequestFilterSensitiveLog;
const GetPartitionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPartitionResponseFilterSensitiveLog = GetPartitionResponseFilterSensitiveLog;
const GetPartitionIndexesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPartitionIndexesRequestFilterSensitiveLog = GetPartitionIndexesRequestFilterSensitiveLog;
const BackfillErrorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BackfillErrorFilterSensitiveLog = BackfillErrorFilterSensitiveLog;
const KeySchemaElementFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.KeySchemaElementFilterSensitiveLog = KeySchemaElementFilterSensitiveLog;
const PartitionIndexDescriptorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PartitionIndexDescriptorFilterSensitiveLog = PartitionIndexDescriptorFilterSensitiveLog;
const GetPartitionIndexesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPartitionIndexesResponseFilterSensitiveLog = GetPartitionIndexesResponseFilterSensitiveLog;
const SegmentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SegmentFilterSensitiveLog = SegmentFilterSensitiveLog;
const GetPartitionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPartitionsRequestFilterSensitiveLog = GetPartitionsRequestFilterSensitiveLog;
const GetPartitionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPartitionsResponseFilterSensitiveLog = GetPartitionsResponseFilterSensitiveLog;
const GetPlanRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPlanRequestFilterSensitiveLog = GetPlanRequestFilterSensitiveLog;
const GetPlanResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPlanResponseFilterSensitiveLog = GetPlanResponseFilterSensitiveLog;
const GetRegistryInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetRegistryInputFilterSensitiveLog = GetRegistryInputFilterSensitiveLog;
const GetRegistryResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetRegistryResponseFilterSensitiveLog = GetRegistryResponseFilterSensitiveLog;
const GetResourcePoliciesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetResourcePoliciesRequestFilterSensitiveLog = GetResourcePoliciesRequestFilterSensitiveLog;
const GluePolicyFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GluePolicyFilterSensitiveLog = GluePolicyFilterSensitiveLog;
const GetResourcePoliciesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetResourcePoliciesResponseFilterSensitiveLog = GetResourcePoliciesResponseFilterSensitiveLog;
const GetResourcePolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetResourcePolicyRequestFilterSensitiveLog = GetResourcePolicyRequestFilterSensitiveLog;
const GetResourcePolicyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetResourcePolicyResponseFilterSensitiveLog = GetResourcePolicyResponseFilterSensitiveLog;
const GetSchemaInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSchemaInputFilterSensitiveLog = GetSchemaInputFilterSensitiveLog;
const GetSchemaResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSchemaResponseFilterSensitiveLog = GetSchemaResponseFilterSensitiveLog;
const GetSchemaByDefinitionInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSchemaByDefinitionInputFilterSensitiveLog = GetSchemaByDefinitionInputFilterSensitiveLog;
const GetSchemaByDefinitionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSchemaByDefinitionResponseFilterSensitiveLog = GetSchemaByDefinitionResponseFilterSensitiveLog;
const SchemaVersionNumberFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SchemaVersionNumberFilterSensitiveLog = SchemaVersionNumberFilterSensitiveLog;
const GetSchemaVersionInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSchemaVersionInputFilterSensitiveLog = GetSchemaVersionInputFilterSensitiveLog;
const GetSchemaVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSchemaVersionResponseFilterSensitiveLog = GetSchemaVersionResponseFilterSensitiveLog;
const GetSchemaVersionsDiffInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSchemaVersionsDiffInputFilterSensitiveLog = GetSchemaVersionsDiffInputFilterSensitiveLog;
const GetSchemaVersionsDiffResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSchemaVersionsDiffResponseFilterSensitiveLog = GetSchemaVersionsDiffResponseFilterSensitiveLog;
const GetSecurityConfigurationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSecurityConfigurationRequestFilterSensitiveLog = GetSecurityConfigurationRequestFilterSensitiveLog;
const SecurityConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SecurityConfigurationFilterSensitiveLog = SecurityConfigurationFilterSensitiveLog;
const GetSecurityConfigurationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSecurityConfigurationResponseFilterSensitiveLog = GetSecurityConfigurationResponseFilterSensitiveLog;
const GetSecurityConfigurationsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSecurityConfigurationsRequestFilterSensitiveLog = GetSecurityConfigurationsRequestFilterSensitiveLog;
const GetSecurityConfigurationsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSecurityConfigurationsResponseFilterSensitiveLog = GetSecurityConfigurationsResponseFilterSensitiveLog;
const GetSessionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSessionRequestFilterSensitiveLog = GetSessionRequestFilterSensitiveLog;
const GetSessionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSessionResponseFilterSensitiveLog = GetSessionResponseFilterSensitiveLog;
const GetStatementRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetStatementRequestFilterSensitiveLog = GetStatementRequestFilterSensitiveLog;
const StatementOutputDataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StatementOutputDataFilterSensitiveLog = StatementOutputDataFilterSensitiveLog;
const StatementOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StatementOutputFilterSensitiveLog = StatementOutputFilterSensitiveLog;
const StatementFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StatementFilterSensitiveLog = StatementFilterSensitiveLog;
const GetStatementResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetStatementResponseFilterSensitiveLog = GetStatementResponseFilterSensitiveLog;
const GetTableRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTableRequestFilterSensitiveLog = GetTableRequestFilterSensitiveLog;
const TableFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TableFilterSensitiveLog = TableFilterSensitiveLog;
const GetTableResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTableResponseFilterSensitiveLog = GetTableResponseFilterSensitiveLog;
const GetTablesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTablesRequestFilterSensitiveLog = GetTablesRequestFilterSensitiveLog;
const GetTablesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTablesResponseFilterSensitiveLog = GetTablesResponseFilterSensitiveLog;
const GetTableVersionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTableVersionRequestFilterSensitiveLog = GetTableVersionRequestFilterSensitiveLog;
const TableVersionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TableVersionFilterSensitiveLog = TableVersionFilterSensitiveLog;
const GetTableVersionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTableVersionResponseFilterSensitiveLog = GetTableVersionResponseFilterSensitiveLog;
const GetTableVersionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTableVersionsRequestFilterSensitiveLog = GetTableVersionsRequestFilterSensitiveLog;
const GetTableVersionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTableVersionsResponseFilterSensitiveLog = GetTableVersionsResponseFilterSensitiveLog;
const GetTagsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTagsRequestFilterSensitiveLog = GetTagsRequestFilterSensitiveLog;
const GetTagsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTagsResponseFilterSensitiveLog = GetTagsResponseFilterSensitiveLog;
const GetTriggerRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTriggerRequestFilterSensitiveLog = GetTriggerRequestFilterSensitiveLog;
const GetTriggerResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTriggerResponseFilterSensitiveLog = GetTriggerResponseFilterSensitiveLog;
const GetTriggersRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTriggersRequestFilterSensitiveLog = GetTriggersRequestFilterSensitiveLog;
const GetTriggersResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTriggersResponseFilterSensitiveLog = GetTriggersResponseFilterSensitiveLog;
const GetUnfilteredPartitionMetadataRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUnfilteredPartitionMetadataRequestFilterSensitiveLog = GetUnfilteredPartitionMetadataRequestFilterSensitiveLog;
const GetUnfilteredPartitionMetadataResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUnfilteredPartitionMetadataResponseFilterSensitiveLog = GetUnfilteredPartitionMetadataResponseFilterSensitiveLog;
const GetUnfilteredPartitionsMetadataRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUnfilteredPartitionsMetadataRequestFilterSensitiveLog = GetUnfilteredPartitionsMetadataRequestFilterSensitiveLog;
const UnfilteredPartitionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UnfilteredPartitionFilterSensitiveLog = UnfilteredPartitionFilterSensitiveLog;
const GetUnfilteredPartitionsMetadataResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUnfilteredPartitionsMetadataResponseFilterSensitiveLog = GetUnfilteredPartitionsMetadataResponseFilterSensitiveLog;
const GetUnfilteredTableMetadataRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUnfilteredTableMetadataRequestFilterSensitiveLog = GetUnfilteredTableMetadataRequestFilterSensitiveLog;
const ColumnRowFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ColumnRowFilterFilterSensitiveLog = ColumnRowFilterFilterSensitiveLog;
const GetUnfilteredTableMetadataResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUnfilteredTableMetadataResponseFilterSensitiveLog = GetUnfilteredTableMetadataResponseFilterSensitiveLog;
const GetUserDefinedFunctionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUserDefinedFunctionRequestFilterSensitiveLog = GetUserDefinedFunctionRequestFilterSensitiveLog;
const UserDefinedFunctionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UserDefinedFunctionFilterSensitiveLog = UserDefinedFunctionFilterSensitiveLog;
const GetUserDefinedFunctionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUserDefinedFunctionResponseFilterSensitiveLog = GetUserDefinedFunctionResponseFilterSensitiveLog;
const GetUserDefinedFunctionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUserDefinedFunctionsRequestFilterSensitiveLog = GetUserDefinedFunctionsRequestFilterSensitiveLog;
const GetUserDefinedFunctionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetUserDefinedFunctionsResponseFilterSensitiveLog = GetUserDefinedFunctionsResponseFilterSensitiveLog;
const GetWorkflowRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkflowRequestFilterSensitiveLog = GetWorkflowRequestFilterSensitiveLog;
const GetWorkflowResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkflowResponseFilterSensitiveLog = GetWorkflowResponseFilterSensitiveLog;
const GetWorkflowRunRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkflowRunRequestFilterSensitiveLog = GetWorkflowRunRequestFilterSensitiveLog;
const GetWorkflowRunResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkflowRunResponseFilterSensitiveLog = GetWorkflowRunResponseFilterSensitiveLog;
const GetWorkflowRunPropertiesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkflowRunPropertiesRequestFilterSensitiveLog = GetWorkflowRunPropertiesRequestFilterSensitiveLog;
const GetWorkflowRunPropertiesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkflowRunPropertiesResponseFilterSensitiveLog = GetWorkflowRunPropertiesResponseFilterSensitiveLog;
const GetWorkflowRunsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkflowRunsRequestFilterSensitiveLog = GetWorkflowRunsRequestFilterSensitiveLog;
const GetWorkflowRunsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkflowRunsResponseFilterSensitiveLog = GetWorkflowRunsResponseFilterSensitiveLog;
const ImportCatalogToGlueRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportCatalogToGlueRequestFilterSensitiveLog = ImportCatalogToGlueRequestFilterSensitiveLog;
const ImportCatalogToGlueResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportCatalogToGlueResponseFilterSensitiveLog = ImportCatalogToGlueResponseFilterSensitiveLog;
const ListBlueprintsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListBlueprintsRequestFilterSensitiveLog = ListBlueprintsRequestFilterSensitiveLog;
const ListBlueprintsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListBlueprintsResponseFilterSensitiveLog = ListBlueprintsResponseFilterSensitiveLog;
const ListCrawlersRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListCrawlersRequestFilterSensitiveLog = ListCrawlersRequestFilterSensitiveLog;
const ListCrawlersResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListCrawlersResponseFilterSensitiveLog = ListCrawlersResponseFilterSensitiveLog;
